### PR TITLE
Update SID only when failed

### PR DESCRIPTION
**Pull request by GuyKh**

### CODE REVIEW

File: electrasmart/cli.py [hunk 1]
```diff
@@ -53,49 +56,72 @@ def list_devices():
...
+        default=None,
+        help="target where baseline status is saved (default: baseline_status_<ac_id>.json)",
+    )
     args = parser.parse_args()
     if args.output_file is None:
         args.output_file = f"baseline_status_{args.ac_id}.json"
     ac = AC(args.imei, args.token, args.ac_id)
     sid = ac.renew_sid()
-    print(f"renewed sid: {sid})
+    print(f"renewed sid: {sid}")
```
Comment (yonatanp) [inline electrasmart/cli.py:69]: :)

**Comment by yonatanp** [comment]
A couple of comments:
1. the `timedelta` object has a `total_seconds()` method, you can use it in `date_diff_in_seconds`
1. regarding the retry feature, it might be useful to discover the specific response that indicates an expired sid, and raise on any other situation. otherwise we might end up double-swamping the server in other error scenarios. also, I would suggest renaming the flag (perhaps `check` instead of `retry`?) as it's not directly describing the function's modified behavior, but rather the caller's context.

Neither of those are reason not to merge, so I'm accepting the PR.

**Comment by GuyKh** [comment]
@yonatanp, will address these issues on Tuesday

**Comment by GuyKh** [comment]
@yonatanp, from what I saw, the discovery of an expired sid is just an empty response.
For example - `GET_DEVICES` command returns the following response -- 
I guess the `status != 0` is a decent indication.

```
{"id":99,"status":1,"desc":null,"data":null}
```

**Pull request merged**